arroyo/types.py

```diff
@@ -85,6 +81,85 @@ def __setstate__(self, state: dict[str, int | datetime]) -> None:
             object.__setattr__(self, slot, value)
 
 
+class BasePayload(Generic[TPayload]):
+    @property
+    def payload(self) -> TPayload:
```

Comment:
why not calling this value (or swap them, Value contains Payload)
I feel having a `Payload` class that contains a payload will be quite confusing.

Reply:
Value seems good to me. It's sufficiently far away from the Kafka concept of value where there's less room for confusion